libs/designsystem/src/lib/components/modal/footer/modal-footer.component.scss

```diff
@@ -20,35 +20,38 @@ ion-footer {
   }
 }
 
-:host-context(.modal-wrapper.full-height) {
-  --kirby-modal-footer-safe-area-bottom: var(--kirby-safe-area-bottom);
-}
-
 :host(.snap-to-keyboard) {
   ion-footer {
     transition: transform utils.$soft-keyboard-transition-leave;
   }
 }
 
+:host(.light) ion-footer {
+  background-color: utils.get-color('background-color');
+}
+
+:host(.inline) ion-footer {
+  background: transparent;
+  box-shadow: none;
+}
+
 :host-context(.keyboard-visible) {
   &.snap-to-keyboard {
     ion-footer {
       transition: transform utils.$soft-keyboard-transition-enter;
+
       // Prevent clean-css from replacing `0px` with `0` or `calc()` won't work:
+
       /* clean-css ignore:start */
       transform: translateY(
         calc((var(--keyboard-offset, 0px) - var(--kirby-modal-footer-safe-area-bottom, 0px)) * -1)
       );
+
       /* clean-css ignore:end */
     }
   }
 }
 
-:host(.light) ion-footer {
-  background-color: utils.get-color('background-color');
-}
-
-:host(.inline) ion-footer {
-  background: transparent;
-  box-shadow: none;
+:host-context(.modal-wrapper.full-height) {
+  --kirby-modal-footer-safe-area-bottom: var(--kirby-safe-area-bottom);
```

Comment:
Do you know which rule triggers this reordering?

Reply:
Good catch. 
Stylelint gave A LOT of errors because of the `no-descending-specificity` rule, but in the end I opted for not creating a too crazy diff, as I feared a lot of the actual changes would fall through the cracks in a large diff. So I disabled the rule instead, for now. 

So these changes are actually me who forgot to revert some of the initial reordering I did.

Reply:
Ah, ok. Fair reasoning. The diff is large enough as it is 🙂

That said, this is one of the rules I was looking forward to using. How about we do a second run in another PR? We can apply `no-descending-specificity`, and perhaps other rules we might want to change, if things doesn't get too crazy too review in one go.